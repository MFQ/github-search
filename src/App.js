import React, { Component } from 'react';
import './App.css';
import SearchBar from "./components/search-bar";
import SearchResults from "./components/search-results";

import { connect } from "react-redux";
import { searchRepo } from "./actions/githubSearchRepo"

const mapStateToProps = (store) => ({
    repos: store.searchRepoReducer.repos,
    fetching: store.searchRepoReducer.fetching,
    fetched: store.searchRepoReducer.fetched,
});

const mapDispatchToProps = (dispatch) => ({
  fetchRepos: (keyword) => dispatch(searchRepo(keyword)),
});

class App extends Component{

  render() {
    const { repos, fetching } = this.props;
    return (
      <div className="App">
        <header className="App-header">
          <SearchBar {...this.props}/>
        </header>
        { fetching && "loading..." }
        <SearchResults repos={repos}/>
      </div>
    );
  }
};

export default connect(mapStateToProps,mapDispatchToProps)(App)
