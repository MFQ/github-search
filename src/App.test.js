import React from 'react';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

import App from "./App";

configure({ adapter: new Adapter() });
const mockStore = configureStore([]);

describe('App', () => {

  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      searchRepoReducer: {
        repos: [],
        fetching: false,
        fetched: false,
      }
    });

    component = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>
    );
  });

  it('should render correctly', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
