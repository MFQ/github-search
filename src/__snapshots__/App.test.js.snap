// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <div>
      <h1>
         Search Github Repos 
      </h1>
      <div
        className="input-group input-group-lg"
      >
        <input
          aria-describedby="inputGroup-sizing-sm"
          aria-label="Large"
          className="form-control"
          onBlur={[Function]}
          onChange={[Function]}
          value=""
        />
      </div>
    </div>
  </header>
  <ul />
</div>
`;
